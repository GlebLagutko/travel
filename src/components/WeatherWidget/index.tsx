import {WeatherIcon} from 'react-open-weather';
import React, {useEffect, useState} from "react";
import {useSelector} from 'react-redux'
import {trackPromise} from 'react-promise-tracker'

const languageState = state => state.value.language;

export default function WeatherWidget({city, cityCoordinates}) {

    const language = useSelector(languageState);

    const [data, setData] = useState({
        weather: false, main: undefined
    });

    useEffect(() => {
        trackPromise(
            fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=da9c9689fa19ac3ed3a29bbd3670acdc&lang=${language}&units=metric`)
                .then(response => response.json()).then(result => {
                setData(result);
            }));

    }, [language]);

    // @ts-ignore
    return (
        <div className='weather-div'>
            <div className='weather-info'>
                <div>{data.weather ? data.main.temp : 0}Â°C</div>
                <div>{data.weather ? data.weather[0].description : 0}</div>
            </div>
            <div className="weather-icon">
                <img src={`https://openweathermap.org/img/wn/${data.weather ? data.weather[0].icon : ''}@2x.png`}
                     style={{height: "40px"}}/>
            </div>
        </div>
    )

};
