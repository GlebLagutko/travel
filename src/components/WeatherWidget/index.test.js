import {render} from '@testing-library/react';

import WeatherWidget from '.';
import React from "react";
import {rest} from 'msw'
import {setupServer} from 'msw/node'
import {Provider} from 'react-redux'
import configureStore from 'redux-mock-store'


describe('ExchangeWidget', () => {

  const response = {
    "coord": {"lon": 10.7461, "lat": 59.9127},
    "weather": [{"id": 800, "main": "Clear", "description": "clear sky", "icon": "01d"}],
    "base": "stations",
    "main": {"temp": 0.99, "feels_like": -4.88, "temp_min": 0, "temp_max": 1.67, "pressure": 1025, "humidity": 47},
    "visibility": 10000,
    "wind": {"speed": 4.12, "deg": 10},
    "clouds": {"all": 0},
    "dt": 1615974029,
    "sys": {"type": 1, "id": 1624, "country": "NO", "sunrise": 1615958854, "sunset": 1616001791},
    "timezone": 3600,
    "id": 3143244,
    "name": "Oslo",
    "cod": 200
  };

  const server = setupServer(
    rest.get('https://api.openweathermap.org/data/2.5/weather?q=Oslo&appid=da9c9689fa19ac3ed3a29bbd3670acdc&lang=en&units=metric', (req, res, ctx) => {
      return res(ctx.json(response))
    }),
  );

  beforeAll(() => server.listen());
  afterAll(() => server.close());
  afterEach(() => server.resetHandlers());

  it('renders correctly', async () => {

    const initialState = {value: {language: "en"}};
    const mockStore = configureStore();
    let store = mockStore(initialState);

    const {container, findByText} = render(
        <Provider store={store}><WeatherWidget city="Oslo"/></Provider>
    );

    expect(await findByText("clear sky")).toBeInTheDocument();
    expect(container).toMatchSnapshot();

  });
});