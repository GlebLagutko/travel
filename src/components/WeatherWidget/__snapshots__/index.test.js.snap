// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeWidget renders correctly 1`] = `
<div>
  <div
    class="weather-div"
  >
    <div
      class="weather-info"
    >
      <div>
        0.99
        Â°C
      </div>
      <div>
        clear sky
      </div>
    </div>
    <div
      class="weather-icon"
    >
      <img
        src="https://openweathermap.org/img/wn/01d@2x.png"
        style="height: 40px;"
      />
    </div>
  </div>
</div>
`;
